### PR TITLE
OS-SPEC: initial commit

### PR DESCRIPTION
**Pull request by philips**

### CODE REVIEW

**Comment by philips** [comment]
Should fix #127. Deprecates #218.

 @cc @cdaylward @alban @jonboulle @vcaputo

**Comment by philips** [comment]
cc @mpasternacki too

**Comment by alban** [comment]
LGTM.

We can add more things later (such as /dev/pts) when the need arises.

**Comment by jonboulle** [comment]
This needs a reference and description from SPEC.md

File: OS-SPEC.md [hunk 1]
```diff
@@ -0,0 +1,23 @@
+# Linux
+
+Linux has both file paths and syscall as part of its ABI. In order to correctly run most applications additional parts of the environment must be setup.
```
Comment (jonboulle) [inline OS-SPEC.md]: The Linux ABI consists of several special file paths and syscalls.
Most applications targeted at Linux will require additional parts of the environment to be set up to run correctly.

File: OS-SPEC.md [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+This document attempts to define a minimal set of things that must exist for most applications.
+
+For ACIs that define the label `os=linux` they can expect this environment by default.
```
Comment (jonboulle) [inline OS-SPEC.md]: s#For ACIs#ACIs#
s#they can#can#

File: OS-SPEC.md [hunk 3]
```diff
@@ -0,0 +1,23 @@
...
+
+## Devices and File Systems
+
+Apps should expect to run in an environment with basic device nodes and filesystems.
```
Comment (jonboulle) [inline OS-SPEC.md]: this is the generic description that should be in the spec, not Linux-specific

File: OS-SPEC.md [hunk 4]
```diff
@@ -0,0 +1,23 @@
...
+The exact device nodes and filesystems available depend on the operating system.
+
+|     Path     |  Type  |
+| ------------ | ------ |
+| /proc        | proc   |
```
Comment (jonboulle) [inline OS-SPEC.md]: can we link to canonical definitions of these?

Comment (philips) [inline OS-SPEC.md]: I can't think of one off the top of my head, @alban ?

File: OS-SPEC.md (hunk 1 above)
Comment (philips) [inline OS-SPEC.md]: Fixed.

File: OS-SPEC.md (hunk 2 above)
Comment (philips) [inline OS-SPEC.md]: fixed

**Comment by philips** [comment]
@jonboulle Reference added to SPEC.md. See https://github.com/appc/spec/pull/221/files#diff-916b3e1e005fd96e3a0546715235477dR664

File: OS-SPEC.md (hunk 4 above)
Comment (jonboulle) [inline OS-SPEC.md]: https://www.kernel.org/doc/Documentation/filesystems/sysfs.txt , https://www.kernel.org/doc/Documentation/filesystems/proc.txt , etc?

File: OS-SPEC.md [hunk 5]
```diff
@@ -0,0 +1,22 @@
...
+
+ACIs that define the label `os=linux` can expect this environment by default.
+
+## Devices and File Systems
+
+|     Path     |  Type  |
+| ------------ | ------ |
+| /proc        | [procfs](https://www.kernel.org/doc/Documentation/filesystems/sysfs.txt) |
+| /sys         | [sysfs](https://www.kernel.org/doc/Documentation/filesystems/proc.txt) |
+| /dev/null    | device                                                                 |
```
Comment (jonboulle) [inline OS-SPEC.md]: https://www.kernel.org/doc/man-pages/online/pages/man4/null.4.html

File: OS-SPEC.md [hunk 6]
```diff
@@ -0,0 +1,22 @@
...
+| /dev/zero    | device                                                                 |
```
Comment (jonboulle) [inline OS-SPEC.md]: https://www.kernel.org/doc/man-pages/online/pages/man4/zero.4.html

File: OS-SPEC.md [hunk 7]
```diff
@@ -0,0 +1,22 @@
...
+| /dev/full    | device                                                                 |
```
Comment (jonboulle) [inline OS-SPEC.md]: https://www.kernel.org/doc/man-pages/online/pages/man4/full.4.html

File: OS-SPEC.md [hunk 8]
```diff
@@ -0,0 +1,22 @@
...
+| /dev/random  | device                                                                 |
```
Comment (jonboulle) [inline OS-SPEC.md]: https://www.kernel.org/doc/man-pages/online/pages/man4/random.4.html

File: OS-SPEC.md [hunk 9]
```diff
@@ -0,0 +1,22 @@
...
+| /dev/urandom | device                                                                 |
```
Comment (jonboulle) [inline OS-SPEC.md]: https://www.kernel.org/doc/man-pages/online/pages/man4/urandom.4.html

File: OS-SPEC.md [hunk 10]
```diff
@@ -0,0 +1,22 @@
...
+| /dev/tty     | device                                                                 |
```
Comment (jonboulle) [inline OS-SPEC.md]: https://www.kernel.org/doc/man-pages/online/pages/man4/tty.4.html

File: OS-SPEC.md [hunk 11]
```diff
@@ -0,0 +1,22 @@
...
+| /dev/console | device                                                                 |
```
Comment (jonboulle) [inline OS-SPEC.md]: https://www.kernel.org/doc/man-pages/online/pages/man4/console.4.html

**Comment by jonboulle** [comment]
LGTM, roll on

**Pull request merged**